src/racer/util.rs

```diff
@@ -79,6 +79,88 @@ pub fn symbol_matches(stype: SearchType, searchstr: &str, candidate: &str) -> bo
     }
 }
 
+pub fn find_closure(src: &str) -> Option<(ByteRange, ByteRange)> {
+    let (pipe_range, _) = closure_valid_arg_scope(src)?;
+    let mut chars = src.chars()
+                       .enumerate()
+                       .skip(pipe_range.end.0)
+                       .skip_while(|(_, c)| {
+                           c.is_whitespace()
+                       });
+    let (start, start_char) = chars.next().map(|(i, c)| (if c == '{' { i + 1 } else { i }, c))?;
+
+    let mut clevel = if start_char == '{' { 1 } else { 0 };
+    let mut plevel = 0;
+
+    let mut chars = chars.peekable();
+    let mut last = None;
+    while chars.peek().is_some() {
```

Comment:
Now looks just `for (i, current) in chars` is enough.

Reply:
MistressMissed itit, sorry!

Reply:
Missed it, sorry!